scipy/spatial/tests/test_kdtree.py

```diff
@@ -721,6 +721,102 @@ def test_kdtree_comparisons():
     assert_equal(sorted(nodes), sorted(nodes[::-1]))
 
 
+def test_ckdtree_build_modes():
+    # check if different build modes for cKDTree give
+    # similar query results
+    np.random.seed(0)
+    n = 5000
+    k = 4
+    points = np.random.randn(n,k)
+    T1 = cKDTree(points).query(points,k=5)[-1]
+    T2 = cKDTree(points,compact_nodes=False).query(points,k=5)[-1]
```

Comment:
the pep8 bot frowns a little at the lack of spaces behind commas in this whole file